src/renderers/dom/test/ReactTestUtilsEntry.js

```diff
@@ -450,6 +450,11 @@ function makeSimulator(eventType) {
       'TestUtils.Simulate expects a component instance and not a ReactElement.' +
         'TestUtils.Simulate will not work if you are using shallow rendering.',
     );
+    invariant(
+      !ReactTestUtils.isCompositeComponent(domComponentOrNode),
+      'TestUtils.Simulate does not support a composite component instance.' +
```

Comment:
There is no space at the line end so sentences will appear to run into each other.

Also, let's phrase it a bit more user friendly. Most people don't know what "composite" means.

For example: "TestUtils.Simulate expected a DOM node as the first argument but received a component instance. Pass the DOM node you wish to simulate the event on instead."

You don't need to mention "DOM component instances" because they don't exist anymore in React. Only DOM node is a valid argument these days.